compiler/parser/src/string.rs

```diff
@@ -14,6 +14,19 @@ pub fn parse_strings(
     let initial_end = values[0].2;
     let initial_kind = (values[0].1 .1 == StringKind::U).then(|| "u".to_owned());
 
+    // Optimization: fast-track the common case of a single string.
+    if values.len() == 1 && matches!(&values[0].1 .1, StringKind::Normal | StringKind::U) {
```

Comment:
If you wanted, you could do `matches!(&*values, [(_, (_, StringKind::Normal | StringKind::U), _)])` for this whole thing (or the equivalent with if-let). I find that using patterns is generally clearer for understanding the structures of tuples over using indices.

Reply:
Nice suggestion, thanks! (I still can't figure out how to take ownership of the `String` itself without the weird `values.into_iter()`, so I couldn't `if-let`.)